File: python/5.Longest-Palindromic-Substring.Expand.py
# 64% faster

class Solution:
    def longestPalindrome(self, s: str) -> str:
        out, cur = "", ""
        for idx, c in enumerate(s):
            p = idx - 1
            q = idx + 1
            cur = c
            while p >= 0 and s[p] == c:
                p -= 1
            while q < len(s) and s[q] == c:
                q += 1
            while p >= 0 and q < len(s):
                if s[p] == s[q]:
                    p -= 1
                    q += 1
                else:
                    break
            cur = s[p+1: q]
            if len(cur) > len(out):
                out = cur
        return out
                
                

File: cpp/5.Longest-Palindromic-Substring.Expand.cpp
// 77% faster

class Solution {
public:
    string longestPalindrome(string s) {
        int p, q;
        int maxL{1};
        int bestp{-1}, bestq{1};
        for(int idx{}; idx < s.length(); idx++){
            p = idx - 1;
            q = idx + 1;
            while(p >= 0 && s[p] == s[idx]){
                p --;
            }
            while(q < s.length() && s[q] == s[idx]){
                q++;
            }
            while(p>=0 && q < s.length() && s[p] == s[q]){
                p--;
                q ++;
            }
            if(q - p - 1 > maxL){
                bestp = p;
                bestq = q;
                maxL = q - p - 1;
            }
        }
        return s.substr(bestp + 1, bestq - bestp - 1);
    }
};